instrumentation/httpurlconnection/library/src/main/java/io/opentelemetry/instrumentation/library/httpurlconnection/HttpUrlInstrumentationConfig.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.library.httpurlconnection;
+
+import io.opentelemetry.instrumentation.api.incubator.semconv.net.PeerServiceResolver;
+import io.opentelemetry.instrumentation.api.internal.HttpConstants;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/** Configuration for automatic instrumentation of HttpURLConnection/HttpsURLConnection requests. */
+public final class HttpUrlInstrumentationConfig {
+    private static List<String> capturedRequestHeaders = new ArrayList<>();
+    private static List<String> capturedResponseHeaders = new ArrayList<>();
+    private static Set<String> knownMethods = HttpConstants.KNOWN_METHODS;
+    private static Map<String, String> peerServiceMapping = new HashMap<>();
+    private static boolean emitExperimentalHttpClientMetrics;
+
+    // Time (ms) to wait before assuming that an idle connection is no longer
+    // in use and should be reported.
+    private static final long CONNECTION_INACTIVITY_TIMEOUT = 10000;
+
+    private HttpUrlInstrumentationConfig() {}
+
+    /**
+     * Configures the HTTP request headers that will be captured as span attributes as described in
+     * <a
+     * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers">HTTP
```

Comment:
better to use permlinks (check all the other links too)

Reply:
Updated! :)